tensorflow/core/kernels/dml_kernel_wrapper.cc

```diff
@@ -128,7 +128,14 @@ void DmlKernelWrapperBase::Compute(OpKernelContext* ctx) {
         const Tensor& input = ctx->input_is_ref(inputIndex)
                                   ? ctx->mutable_input(inputIndex, false)
                                   : ctx->input(inputIndex);
-        ctx->set_output(outputIndex, input);
+
+        Tensor output;
+        // Copies underlying data pointer, but uses the output shape provided.
+        if (!output.CopyFrom(input, output_shapes[outputIndex])) {
+          ctx->SetStatus(errors::Internal("Error during copy."));
+        }
```

Comment:
```suggestion
        OP_REQUIRES(ctx, output.CopyFrom(input, output_shapes[outputIndex]));
```

Just setting the status here won't enough and we will still attempt to call `ctx->set_output(outputIndex, output);` OP_REQUIRES sets the error status but also returns from the function.